src/lib/core/utils/data-element-constraints.ts

```diff
@@ -133,9 +140,18 @@ export function flattenConstraints(
       const shapeIsValid =
         isEmpty(constraint.allowedShapes) ||
         constraint.allowedShapes?.includes(variable.dataShape!);
+      const passesMaxValuesConstraint =
+        constraint.maxNumValues === undefined ||
+        variable.distinctValuesCount === undefined ||
```

Comment:
This can also be removed, due to the type narrowing, if the changes in `src/lib/core/types/study.ts` are undone.